### PR TITLE
Events for entity creation and destruction

### PR DESCRIPTION
**Pull request by OnigiriOni**
This is for issue #22:
Mods can notify other mods if they change something in script with certain events. This change implements this feature to be in line with the base-game event system. Since aircraft takeoffs and landings are a rare occurrence, this should not decrease performance at all.

Already self-tested and self-verified.

### CODE REVIEW

**Comment by OnigiriOni** [comment]
This is for issue #22, don't know if I could have linked it somewhere.

File: changelog.txt [hunk 1]
```diff
@@ -1,4 +1,9 @@
 ---------------------------------------------------------------------------------------------------
+Version: 1.5.6
+Date: 2023-11-22
+  Changes:
+    - Added events on entity creation and destruction which other mods can subscribe to
```
Comment (jaihysc) [inline changelog.txt:5]:
```suggestion
    - Added events on entity creation and destruction events when plane transitions between airborne and grounded which other mods can subscribe to
```

**Review by jaihysc — COMMENTED** [review]
PR looks fine, there is another location where planes can be destroyed - will this be an issue for you?

https://github.com/jaihysc/Factorio-AircraftRealism/blob/f465cb46845895efe815dcd198999c354cf04bbc/logic/planeRunway.lua#L28

**Comment by OnigiriOni** [comment]
The .die() part you mentioned should be fine. I had no problems so far with just the changes I made.

Here is the description for LuaEntity::die() from the API Docs:
desc

Its just that LuaEntity::destroy() LuaSurface::create_entity() do not raise events per default and that can create some holes when everything else does fire events.

**Comment by OnigiriOni** [comment]
After some testing where I drove the Jet to death on a sandbank I can definitely confirm that everything is working as intended.

**Comment by jaihysc** [comment]
Sounds good, I made the new release

**Pull request merged**